test/reduce.jl

```diff
@@ -16,6 +17,7 @@
 @test Base.mapfoldl((x)-> x ⊻ true, |, [true false true false false]) == true
 @test Base.mapfoldl((x)-> x ⊻ true, |, false, [true false true false false]) == true
 
+@test foldr(+, Int64[]) == 0
```

Comment:
In the `foldr` test I can just reference issue #20144 and PR #20160 (this one), but on the `foldl` test I can only reference issue #7465 since there is no PR.
@Sacha0, do I just reference the relevant issue for the `foldl` test or do I add the SHA of the relevant commit? (b674eed7dacb961c2b8800bd0abe8ea9a2da25de)